http/server_test.ts

```diff
@@ -298,4 +303,99 @@ test(async function writeStringReaderResponse(): Promise<void> {
   assertEquals(decoder.decode(line), "0");
 });
 
+function createConnMock(): Deno.Conn {
+  return {
+    localAddr: "",
+    remoteAddr: "",
+    rid: -1,
+    closeRead: (): void => {},
+    closeWrite: (): void => {},
+    read: async (): Promise<Deno.ReadResult> => {
+      return { eof: true, nread: 0 };
+    },
+    write: async (): Promise<number> => {
+      return -1;
+    },
+    close: (): void => {}
+  };
+}
+
+test(async function readRequestError(): Promise<void> {
+  let input = `GET / HTTP/1.1
+malformedHeader
+`;
+  const buf = new Buffer(enc.encode(input));
+  const reader = new BufReader(buf);
+  const conn = createConnMock();
+  const [_, err] = await readRequest(conn, reader);
+  const e: any = err; // eslint-disable-line @typescript-eslint/no-explicit-any
+  assert(e instanceof Error);
+  assertEquals(e.message, "Unable to proceed request");
+});
+
+// Port from Go
+// https://github.com/golang/go/blob/master/src/net/http/request_test.go
```

Comment:
Link to the line number using a commit in the link.
Example:  https://github.com/golang/go/blob/1d1ba8/src/net/http/request_test.go#L380-L446